File: frontend/test_simple.js
// syntax error
// function brokenFunction(
//     console.log("This function has missing parenthesis"

function testAddition() {
    let result = 2 + 2;
    if (result !== 4) throw new Error("Addition failed");
    console.log("Addition test passed!");
}

function testSubtraction() {
    let result = 5 - 3;
    if (result !== 2) throw new Error("Subtraction failed");
    console.log("Subtraction test passed!");
}

// // faild unit test
// function testMultiplication() {
//     let result = 3 * 3;
//     if (result !== 10) throw new Error("Multiplication failed");
//     console.log("This won't print because test fails before here");
// }

File: backend/tests/test_simple.py
# syntax error
# def broken_function(
#     print("This function has missing parenthesis"

# success unit test    
def test_addition():
    result = 2 + 2
    assert result == 4
    
def test_subtraction():
    result = 5 - 3
    assert result == 2

#faild unit test
# def test_multiplication():
#     result = 3 * 3
#     assert result == 10

# if __name__ == "__main__":
#     test_addition()
#     test_subtraction()
#     # test_multiplication()
#     print("All tests completed!")